jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MutableHttpFields.java

```diff
@@ -67,7 +68,17 @@ protected MutableHttpFields(int capacity)
      */
     protected MutableHttpFields(HttpFields fields)
     {
-        add(fields);
+        if (fields instanceof ImmutableHttpFields immutable)
+        {
+            _immutable = true;
+            _fields = immutable._fields;
+            _size = immutable._size;
+        }
+        else
+        {
+            _fields = new HttpField[INITIAL_SIZE];
```

Comment:
You know the size of the fields, so you may use that instead of `INITIAL_SIZE`, no?